### PR TITLE
Add react/preact as peer-dependencies to react-emotion/preact-emotion

### LINKED ISSUE #684: react-emotion is missing react (peer)dependency
There is a direct dependency on react in your code but there is no reference to it in package.json

https://github.com/emotion-js/emotion/blob/d9a9e332dfbc13ea78325082393ccdd54d34afd6/packages/react-emotion/src/index.js#L2

https://github.com/emotion-js/emotion/blob/d9a9e332dfbc13ea78325082393ccdd54d34afd6/packages/react-emotion/package.json#L21-L27

This would normally not be an issue when using a package manager (NPM) that optimizes the resulting dependency tree since react will be placed at the root of the node_modules directory. Otherwise you'll get an error like me:

> Module not found: Error: Can't resolve 'react' in ...

### PR DESCRIPTION
**Pull request by mgroenhoff**
Fixes #684

### CODE REVIEW

File: packages/react-emotion/package.json [hunk 1]
```diff
@@ -23,7 +23,8 @@
     "create-emotion-styled": "^9.1.2"
   },
   "peerDependencies": {
-    "emotion": "^9.1.3"
+    "emotion": "^9.1.3",
+    "react": ">= 15"
```
Comment (Andarist) [inline packages/react-emotion/package.json]: Those `>=` ranges are too wide, we can't guarantee being compatible with future versions of react/preact, so it's "dangerous" to make this assumption here. If we are gonna be compatible with future versions then we should loosen up this range in the future to include next versions. 

Let's just stick to something like `15.x || 16.x` here and `8.x` for `preact`

Comment (mgroenhoff) [reply packages/react-emotion/package.json]: Allright. I think these need to be changed these as well then.
https://github.com/emotion-js/emotion/blob/d9a9e332dfbc13ea78325082393ccdd54d34afd6/packages/create-emotion-styled/package.json#L24
https://github.com/emotion-js/emotion/blob/d9a9e332dfbc13ea78325082393ccdd54d34afd6/packages/emotion-theming/package.json#L50-L51

Comment (Andarist) [reply packages/react-emotion/package.json]: yeah, imho they should be too, good catch 👍

File: packages/create-emotion-styled/package.json [hunk 2]
```diff
@@ -21,13 +21,13 @@
     "@emotion/is-prop-valid": "^0.6.1"
   },
   "peerDependencies": {
-    "prop-types": ">= 15"
+    "prop-types": "^15.6.1"
```
[the remark this answers is not in the document]
Comment (mgroenhoff) [reply packages/create-emotion-styled/package.json]: Right, will do. I did not see where the peer dependency came from by looking at the source so I thought it did not matter but now i see it is actually emitted in the bundle.

**Pull request merged**